app/setup.php

```diff
@@ -74,6 +73,73 @@
      */
     add_theme_support('customize-selective-refresh-widgets');
 
+    /**
+     * Gutenberg support
+     * @link https://wordpress.org/gutenberg/handbook/extensibility/theme-support/
+     */
+    add_theme_support('editor-styles');
+    add_theme_support('wp-block-styles');
+    add_theme_support('align-wide');
+    add_theme_support('disable-custom-colors');
+    add_theme_support('disable-custom-font-sizes');
+    add_theme_support('responsive-embeds');
+
+    add_theme_support('editor-color-palette', [
+        [
+            'name' => __('Primary', 'sage'),
+            'slug' => 'primary',
+            'color' => tailwind('theme.colors.primary'),
+        ],
+        [
+            'name' => __('Secondary', 'sage'),
+            'slug' => 'secondary',
+            'color' => tailwind('theme.colors.secondary'),
+        ],
+        [
+            'name' => __('Black', 'sage'),
+            'slug' => 'black',
+            'color' => tailwind('theme.colors.black'),
+        ],
+        [
+            'name' => __('White', 'sage'),
+            'slug' => 'white',
+            'color' => tailwind('theme.colors.white'),
+        ],
+    ]);
+
+    add_theme_support('editor-font-sizes', [
+        [
+            'name' => __('xsmall', 'sage'),
+            'shortName' => __('XS', 'sage'),
+            'slug' => 'xs',
+            'size' => tailwind('theme.fontSize.xs'),
+        ],
+        [
+            'name' => __('small', 'sage'),
+            'shortName' => __('S', 'sage'),
+            'slug' => 'sm',
+            'size' => tailwind('theme.fontSize.sm'),
+        ],
+        [
+            'name' => __('regular', 'sage'),
+            'shortName' => __('R', 'sage'),
+            'slug' => 'base',
+            'size' => tailwind('theme.fontSize.base'),
+        ],
+        [
+            'name' => __('large', 'sage'),
+            'shortName' => __('L', 'sage'),
+            'slug' => 'xl',
+            'size' => tailwind('theme.fontSize.xl'),
+        ],
+        [
+            'name' => __('xlarge', 'sage'),
+            'shortName' => __('XL', 'sage'),
+            'slug' => 'xxl',
```

Comment:
maybe change back to 2xl as tailwind uses. Problem is wordpress adds a dash in between numbers and characters but `tailwindcss-gutenberg` should fix this.

Reply:
Also another limitation here is that `tailwind()` php helper wont find the dynamically generated tailwind variables in `theme.gutenberg` feels like these should point to the same.

Another option would be to read the tailwind values from a json file rather than have the json file generated.

Reply:
Yes, it might be cleaner. But that adds a third config file basically...